### DISCUSSION TITLE
First/last data points on X-axis

### DISCUSSION THREAD

**Discussion by nullsafety** [patrykandpatrick/vico#947] [Questions]
### How to reproduce

I have an issue similar to previous ones, such as [this one](https://github.com/patrykandpatrick/vico/discussions/394). However, my problem is not that the last label the library tries to render is missing—it's that the library doesn't attempt to render the last point at all. I've also tried adjusting offsets, but that didn't help. As I understand it, in the current version of the library, the default padding is configured to ensure that extreme values are displayed without being replaced by "...".

I attempted to add points manually by implementing my own `ItemPlacer`. I used the code from `AlignedHorizontalAxisItemPlacer` and added mandatory rendering of the last data point in `getLabelValues`, which just looked like this:
`labelValues.add(fullXRange.endInclusive)`

However, the data is still displayed as "..." instead of showing the actual values.

I also assumed that the issue might be related to the step size and tried to make it consistent. But when I did this, all points were displayed as "...".

It seems like these labels need to be synchronized with some other data, but I can't figure out what they need to be synchronized with.

### Observed behavior

<img width="409" alt="Снимок экрана 2024-11-06 в 22 06 00" src="https://github.com/user-attachments/assets/cbe624c5-fe35-4658-974a-45e8699cbd7a">
<img width="258" alt="Снимок экрана 2024-11-06 в 22 07 24" src="https://github.com/user-attachments/assets/c9bc2209-289c-45c6-9a6b-52fae8b9cb31">

### Expected behavior

Like this but in case 03.11 is last point in the data:
<img width="417" alt="Снимок экрана 2024-11-07 в 02 13 22" src="https://github.com/user-attachments/assets/6b9bee08-1396-429a-9732-b272b8d9056f">

### Vico version(s)

2.0.0-beta.2

### Android version(s)

14

**Comment by Gowsky** [comment]
Hello, @nullsafety. First, this seems to be a question and not a bug report, so I'm converting it to a discussion. Please see [here](https://patrykandpatrick.com/vico/wiki/contributing#issues-and-discussions).

Second, we need to see your code to assist you. Please share a runnable example that demonstrates the behavior you're referring to.
